src/transformers/modeling_outputs.py

```diff
@@ -1622,3 +1623,40 @@ class SampleTSPredictionOutput(ModelOutput):
     """
 
     sequences: torch.FloatTensor = None
+
+
+@dataclass
+class MaskedImageModelingOutput(ModelOutput):
+    """
+    Base class for outputs of masked image completion / in-painting models.
+
+    Args:
+        loss (`torch.FloatTensor` of shape `(1,)`, *optional*, returned when `bool_masked_pos` is provided):
```

Comment:
The "bool_masked_pos" argument is highly specific to the xxxForMaskedImageModeling models we have, so I assume this output class is only meant for them?

Otherwise if we plan to use this output class for other models in the future, we may have to remove that in the docstring

Reply:
The image in-painting / completion models will require bool_masked_pos + the ground truth image (as the input is masked) to compute a loss so I don't think there will be any changes needed, or at least it'll be very minimal